pennylane/gradients/parameter_shift.py

```diff
@@ -23,6 +23,40 @@
 from .finite_difference import finite_diff, generate_shifted_tapes
 
 
+NONINVOLUTORY_OBS = {
+    "Hermitian": lambda obs: obs.__class__(obs.matrix @ obs.matrix, wires=obs.wires),
+    "SparseHamiltonian": lambda obs: obs.__class__(obs.matrix @ obs.matrix, wires=obs.wires),
+    "Projector": lambda obs: obs,
+}
+"""Dict[str, callable]: mapping from a non-involutory observable name
+to a callable that accepts an observable object, and returns the square
+of that observable.
+"""
+
+
+def _square_observable(obs):
+    """Returns the square of an observable."""
+
+    if isinstance(obs, qml.operation.Tensor):
+        # Observable is a tensor, we must consider its
+        # component observables independently. Note that
+        # we assume all component observables are on distinct wires.
+
+        components_squared = []
+
+        for comp in obs.obs:
+
+            try:
+                components_squared.append(NONINVOLUTORY_OBS[comp.name](comp))
+            except KeyError:
+                # component is involutory
+                pass

```

Comment:
Looks like this line is not being covered.

Reply:
Added a test for this!